### PR TITLE
Update spring-boot version to 2.4.6 and other dependencies

### PR DESCRIPTION
**Pull request by DanielFran**
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/templates/gradle.properties.ejs [hunk 1]
```diff
@@ -57,19 +61,19 @@ cassandraDriverVersion=4.9.0
 
 # gradle plugin version
 jibPluginVersion=<%= JIB_VERSION %>
-gitPropertiesPluginVersion=2.2.4
```
[the remark this answers is not in the document]
Comment (DanielFran) [reply generators/server/templates/gradle.properties.ejs:60]: Yes I remember when I try updating to 2.3.0... Just waiting that versions are available in maven server to validate it

**Comment by mraible** [comment]
Now that [Spring Boot 2.5](https://spring.io/blog/2021/05/20/spring-boot-2-5-is-now-ga) is out, maybe that should be a priority over this upgrade?

**Comment by DanielFran** [comment]
It is my next step

**Comment by DanielFran** [comment]
@mraible Actually CI is too unstable to try 2.5.0
Just waiting that issue with eslint is fixed to try the upgrade

**Comment by mraible** [comment]
@DanielFran If there's anything I can do to help, please let me know.

**Comment by DanielFran** [comment]
No, I'm in contact with @mshima with his workaround https://github.com/jhipster/generator-jhipster/pull/15059 to temporally merge it to test this PR and next with spring-boot 2.5.0.
Too lazy to test all combination locally... :P

**Pull request merged**